Use the m-lab/go/host package to parse -hostname flag

nkinkade opened the pull request:
This provides a consistent interface for processing a hostname, and allows us to handle special cases in the `host` package instead of in this code. For example, the `host` package knows about managed instance group random suffixes on instance/node names, and can handle that gracefully for uuid-annotator.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/m-lab/uuid-annotator/50)
<!-- Reviewable:end -->

nkinkade committed:
Use the m-lab/go/host package to parse -hostname flag

This provides a consistent interface for processing a hostname, and
allows us to handle special cases in the `host` package instead of in
this code. For example, the `host` package knows about managed instance
group random suffixes on instance/node names, and can handle that
gracefully for uuid-annotator.

stephen-soltesz reviewed (approved):
With suggested change. <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/m-lab/uuid-annotator/50#-:-NOGpyuPDfr6Nwe-OJQJ:bbsitr4)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @nkinkade)

___
*[`main.go` line 87 at r1](https://reviewable.io/reviews/m-lab/uuid-annotator/50#-NOGpabz0TPOZ41d4WDa:-NOGpac-5rnt7GsIzB7N:b-122s59) ([raw file](https://github.com/m-lab/uuid-annotator/blob/99ea1ad00dab67fdccfaa5cb25b4b62816c546bb/main.go#L87)):*
> ```Go
> 	h, err := host.Parse(*hostname)
> 	rtx.Must(err, "Failed to parse -hostname flag value")
> 	mlabHostname := fmt.Sprintf("%s-%s.%s.%s", h.Machine, h.Site, h.Project, h.Domain)
> ```

Prefer `h.String()` which reconstructs the hostname but does not include the suffix.

https://pkg.go.dev/github.com/m-lab/go@v0.1.57/host#Name.String


<!-- Sent from Reviewable.io -->

nkinkade committed:
Uses builtin String() func instead of fmt

The m-lab/go/host package has a built-in String() function which returns
the hostname, minus any suffix. Use this instead of some long
fmt.Sprintf() call.

nkinkade reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/m-lab/uuid-annotator/50)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @stephen-soltesz)

___
*[`main.go` line 87 at r1](https://reviewable.io/reviews/m-lab/uuid-annotator/50#-NOGpabz0TPOZ41d4WDa:-NOGrkj65chG_gFwuXdD:bexn9y3) ([raw file](https://github.com/m-lab/uuid-annotator/blob/99ea1ad00dab67fdccfaa5cb25b4b62816c546bb/main.go#L87)):*
<details><summary><i>Previously, stephen-soltesz (Stephen Soltesz) wrote…</i></summary><blockquote>

Prefer `h.String()` which reconstructs the hostname but does not include the suffix.

https://pkg.go.dev/github.com/m-lab/go@v0.1.57/host#Name.String
</blockquote></details>

Thank you. Done. It's rather embarrassing when someone else has to point out functionality in a package that you yourself wrote at some point in the past.


<!-- Sent from Reviewable.io -->
